radix-engine/src/model/component/substates.rs

```diff
@@ -32,3 +33,9 @@ impl ComponentInfoSubstate {
 pub struct ComponentRoyaltyConfigSubstate {
     pub royalty_config: RoyaltyConfig,
 }
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[scrypto(TypeId, Encode, Decode)]
+pub struct ComponentRoyaltyAccumulatorSubstate {
+    pub royalty: Resource, // TODO: wrap with a vault?
```

Comment:
Yes, definitely should wrap with vault

Reply:
Yeah, I can definitely see the point -  all the resource are stored in vaults. 

The only concern I had was - do you need the extra functionality provided by vaults? Do we accept the additional cost associated with an extra hop? Is there any problem allowing component accessing royalty the same way accessing vaults?